### DISCUSSION TITLE
https://turboviplay.com/

### DISCUSSION THREAD

**Discussion by PCL86** [aclap-dev/video-downloadhelper#845] [Help and Q&A]
### Your question

I would like you to support video downloads from [https://emturbovid.com/](https://turboviplay.com/)
For example, when I select TV as the server for https://supjav.com/278511.html, I get a download error
I would like you to make this possible

### Addon & CoApp version and details

SideDownload error:

[out#0/mp4 @ 000002910ed98d00] Output file does not contain any stream

Error opening output file E:\Video\Caribbeancom 060824-001 Full Power SEX Miyu Morita .mp4.

Error opening output files: Invalid argument

**Comment by Mattmjs** [comment]
Try the other servers which do work.

**Comment by GrampaWildWilly** [comment]
I do not work for VDH.  I'm just another user.

I visited your web page at supjav.com.  I discovered that this is a porn site.  My VDH menu showed a handful of videos I could download, each of duration 6 seconds.  Clearly, those are useless.  Time for Plan B.

I was able to find an HLS master manifest in the Network Monitor.  I was able to download 10 minutes of this with ffmpeg.  Here is the log of that download:

[Live Porn Stream mp4 Log.txt](https://github.com/user-attachments/files/15751469/Live.Porn.Stream.mp4.Log.txt)

A few lines from the top, you'll find one line that looks like this:

`  Duration: N/A, start: 1140.268233, bitrate: 0 kb/s`

The N/A indication for the duration is a tipoff that this is a livestream.  The start value is the number of seconds the stream had been live before I arrived & issued my ffmpeg command.  Actually, that may not be 100% true.  That may just be an indication of how much of the stream has been archived on the web site prior to the moment I arrived.  Either way, it tells us how far into the past we are able to go back to record this livestream.  The ffmpeg parameter `-live_start_index 0` you can see in my ffmpeg command rewinds the recording to its start.  The ffmpeg parameter `-t 00:10:00` says to record 10 minutes of this.

If you are sharp-eyed you will notice that a manifest at URL `https://b-hls-20.sacdnssedge.com/hls/119433068/119433068_480p.m3u8` is being read repeatedly.  This is indicated on the lines that look like this:

`Opening 'https://b-hls-20.sacdnssedge.com/hls/119433068/119433068_480p.m3u8' for reading`

Now look for the lines that begin with the string `size=`.  In each of those lines, look for the string `time=`.  That gives you the time index of the block of the livestream that ffmpeg has just read & saved on disc.  This looks to me like every 2 seconds, this web site is rewriting the contents of the manifest at that URL.  That's why ffmpeg keeps rereading it.  The manifest appears to contain information for only 2 seconds of the livestream.  After ffmpeg reads those 2 seconds worth of video, it sees that the manifest has been updated.  So ffmpeg reads the manifest again & gets the next 2 seconds worth of this livestream.  And so on until the livestream ends.  I didn't care to wait that long.  I decided to just get the first 10 minutes as a proof of concept.

Now scroll down near the bottom of the log, where the 10 minutes have been recorded & ffmpeg stops, as I told it to do.  You'll see that the average download speed for this livestream was barely 155K bytes per second.  That is a horribly slow speed, especially given that this is only of resolution 854x480.  There does not seem to be any choice here.  There are no variants of any other, neither better nor worse, resolution on offer here.  I was not downloading the stream as it was happening.  I was downloading the beginning of the stream, something that was already saved on the web site over an hour earlier.  The web site still refused to give me this stream any faster than it would have taken to just sit & watch it.  If I had been trying to record the live part of the livestream, I could understand that slow speed.  But I was getting the part that was old & already saved.  In my opinion, that web site sucks.  Understand, this is an evaluation of the technology, not the content.

I played this MP4 in VLC.  It played fine with both video & audio.  I didn't sit & watch it, just skimmed it, sampling it at intervals.  There didn't appear to be any problems with it.

This is not the first time somebody has reported a problem with VDH recording a livestream.  Different web sites do things differently.  VDH can record some livestreams but not others.  It seems this is one of those others.  You have reported this.  Now it's up to @paulrouget Paul to decide whether he's going to add support for this style of livestream.  Will he do it?  I can't say.  Like I said at the start, I don't work for VDH.  So you have a choice.  You can wait until VDH can handle this web site.  That may be a long wait.  That may be an infinite wait if Paul decides he won't address this.  On the other hand, you can use ffmpeg right this minute to download whatever you want off this site.  It's up to you which choice you make.
